### PR TITLE
Solitaire: Adds `space` key for drawing cards, `tab` key for auto move eligible cards to foundation stacks

### LINKED ISSUE #7251: Solitaire: Implement an action to move all eligible cards to the foundation stacks
I vaguely remember Windows Solitaire having that when you press the `space` key.

### PR DESCRIPTION
**Pull request by matthewbjones**
Updates the Solitaire game to add two new key events:

1. Pressing `space` key will draw new card(s), the same as clicking on the draw pile
2. Pressing `tab` key will have the game attempt to move all eligible cards from the tableau stacks and play stack to the foundation stacks

Fixes #7251

### CODE REVIEW

**Comment by gunnarbeutner** [comment]
Very nice! I did find a bug however: When moving a card from the play stack to the foundation stacks with `tab` I'm no longer able to move the card that is drawn next.

**Comment by matthewbjones** [comment]
> Very nice! I did find a bug however: When moving a card from the play stack to the foundation stacks with `tab` I'm no longer able to move the card that is drawn next.

Good catch! I've updated the PR to introduce `pop_waste_to_play_stack()` which resolves the issue

File: Userland/Games/Solitaire/Game.cpp [hunk 1]
```diff
@@ -144,7 +152,12 @@ void Game::keydown_event(GUI::KeyEvent& event)
 
     if (event.shift() && (event.key() == KeyCode::Key_F12))
         start_game_over_animation();
-    else if (event.shift() && (event.key() == KeyCode::Key_F11))
+    else if (event.key() == KeyCode::Key_Tab)
+        auto_move_eligible_cards_to_stacks();
+    else if (event.key() == KeyCode::Key_Space) {
+        draw_cards();
+        invalidate_layout(); // FIXME: Stock stack won't render properly after draw_cards() without this
+    } else if (event.shift() && (event.key() == KeyCode::Key_F11))
```
Comment (trflynn89) [inline Userland/Games/Solitaire/Game.cpp:160]: I think when adding parentheses to one of the `else if` blocks, we should add them to the other blocks in the chain as well.

Comment (matthewbjones) [reply Userland/Games/Solitaire/Game.cpp:160]: Did you mean curly braces and not parentheses?

Comment (trflynn89) [reply Userland/Games/Solitaire/Game.cpp:160]: Oops :facepalm: I did mean braces, sorry

Comment (matthewbjones) [reply Userland/Games/Solitaire/Game.cpp:160]: No worries! I've updated the code to use curly braces, and will stick to that coding style going forward. Thanks!

**Review by trflynn89 — APPROVED** [review]

File: Userland/Games/Solitaire/Game.cpp (hunk 1 above)
Comment (linusg) [reply Userland/Games/Solitaire/Game.cpp:160]: This doesn't seem to be resolved - this should be:

```c++
} else if (event.shift() && (event.key() == KeyCode::Key_F11)) {
    dump_layout();
}
```

i.e. if any of the if/else blocks needs curly braces, all of them should get them. The parentheses around the key equality comparison are not needed though :)

**Review by linusg — CHANGES_REQUESTED** [review]

File: Userland/Games/Solitaire/Game.cpp (hunk 1 above)
Comment (matthewbjones) [reply Userland/Games/Solitaire/Game.cpp]: Thanks, good catch on that last curly brace, overlooked it when I was updating the others. 🤦

The parentheses for key comparison were existing to the codebase but I have cleaned that up while I'm editing these lines anyways.

**Comment by linusg** [comment]
Great work on all of these PRs btw! :tada:

**Comment by linusg** [comment]
This has conflicts with the other solitaire PR, sorry about that!

**Comment by matthewbjones** [comment]
> This has conflicts with the other solitaire PR, sorry about that!

No worries, I figured that would happen when I was working on it :-) Should be resolved now.

**Pull request merged**